diff --git a/src/PVRIptvData.cpp b/src/PVRIptvData.cpp
index cc106de..06395b6 100644
--- a/src/PVRIptvData.cpp
+++ b/src/PVRIptvData.cpp
@@ -22,15 +22,18 @@
  *
  */
 
-#include <sstream>
-#include <string>
+#include <ctime>
 #include <fstream>
 #include <map>
+#include <sstream>
 #include <stdexcept>
+#include <string>
+
 #include "zlib.h"
 #include "rapidxml/rapidxml.hpp"
 #include "PVRIptvData.h"
 #include "p8-platform/util/StringUtils.h"
+#include "client.h"
 
 #define M3U_START_MARKER        "#EXTM3U"
 #define M3U_INFO_MARKER         "#EXTINF"
diff --git a/src/PVRIptvData.h b/src/PVRIptvData.h
index b147650..b87dfdc 100644
--- a/src/PVRIptvData.h
+++ b/src/PVRIptvData.h
@@ -24,8 +24,8 @@
  */
 
 #include <vector>
-#include "p8-platform/util/StdString.h"
-#include "client.h"
+#include "p8-platform/os.h"
+#include "libXBMC_pvr.h"
 #include "p8-platform/threads/threads.h"
 
 struct PVRIptvEpgEntry
diff --git a/src/client.cpp b/src/client.cpp
index 7cd838f..bce87b0 100644
--- a/src/client.cpp
+++ b/src/client.cpp
@@ -23,7 +23,6 @@
  */
 
 #include "client.h"
-#include "xbmc_pvr_dll.h"
 #include "PVRIptvData.h"
 #include "p8-platform/util/util.h"
 
